src/core/EventBus.js

```diff
@@ -49,11 +51,13 @@ function EventBus() {
 
         var handler = {
             callback: listener,
-            scope: scope
+            scope: scope,
+            priority: priority !== undefined ? priority : LOW_PRIORITY
         };
 
         handlers[type] = handlers[type] || [];
         handlers[type].push(handler);
+        handlers[type].sort(function (a, b) {return a.priority - b.priority;});
```

Comment:
Do we want to do a manual insertion in order instead of a full sort? It's algorithmically faster, but that's really not my concern here.

By doing a full `.sort` we're using sort methods that are not guaranteed to be stable. This shouldn't be a big deal, but could lead to hard to track down issues where events are not fired in a consistent order. And since we know we'll always have an empty or sorted array inserting in order seems like a good way to avoid the sort stability uncertainty.